### PR TITLE
Added SubGraphOp support in compare_function

### PR DESCRIPTION
**Pull request by GlebKazantaev**
* Added SubGraphOp support in copmare_functoins
* Added tests with TI

### CODE REVIEW

**Review by ilyachur — APPROVED** [review]

File: inference-engine/tests/functional/inference_engine/transformations/compare_functions_test.cpp [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+//
+
+#include <gtest/gtest.h>
+
+#include "common_test_utils/test_common.hpp"
+#include <string>
+#include <memory>
+#include <queue>
+
+#include <ngraph/pass/manager.hpp>
```
Comment (pelszkow) [inline inference-engine/tests/functional/inference_engine/transformations/compare_functions_test.cpp:12]: This includes looks like local so pleas use `"..."` to include such headers.

File: inference-engine/tests/functional/inference_engine/transformations/compare_functions_test.cpp [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+#include <transformations/utils/utils.hpp>
+#include <transformations/init_node_info.hpp>
+
+#include "common_test_utils/ngraph_test_utils.hpp"
+
+using namespace testing;
+using namespace ngraph;
+
+TEST(TransformationTests, CompareFunctoinsTIPositive) {
+    std::shared_ptr<ngraph::Function> f(nullptr), f_ref(nullptr);
```
Comment (pelszkow) [inline inference-engine/tests/functional/inference_engine/transformations/compare_functions_test.cpp:24]: IMHO, we should use [IIFE](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-lambda-init) for such complex initialization.

**Review by pelszkow — APPROVED** [review]

File: inference-engine/tests/functional/inference_engine/transformations/compare_functions_test.cpp (hunk 1 above)
Comment (GlebKazantaev) [reply inference-engine/tests/functional/inference_engine/transformations/compare_functions_test.cpp:12]: It's a public header which is suggested to use when you use pass::Manager.

File: inference-engine/tests/functional/inference_engine/transformations/compare_functions_test.cpp (hunk 2 above)
Comment (GlebKazantaev) [reply inference-engine/tests/functional/inference_engine/transformations/compare_functions_test.cpp:24]: Yes, I agree. I'll do it in a separate PR not to block merge of this one.

**Pull request merged**